text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,248 @@
+  responsible for estimate this effort.
+- _“Strategic alignment”_ - prioritize based on AWS / department / team goals. We will use our weekly, monthly, and
+  quarterly planning meetings to identify modules that meet this criteria. For example, in 2019, there was a Developer
+  Tools strategic initiative to enable containerized and serverless application workloads which informed our decisions
+  on what construct library modules to prioritize for CDKs GA launch.
+- _“Focus on what customers are actually using”_ - prioritize constructs that only have CFN resources, but are highly
+  utilized (according to analytics data). A highly used, CFN-only construct library modules might indicate that the
+  low-level resources are sufficient for customer’s primary use cases, and high-level abstractions are not necessary.
+
+# Appendix A: GitHub artifacts
+
+This diagram illustrates the different GitHub artifacts and their relationships.
+
+- **module tracking issue** - module tracking issues represent an entire AWS service and summarize the planned scope of
+  work for a specific construct library module. This is the issue we want customers to +1 to help us prioritize work on
+  modules that are currently CFN-only, and is the issue we will place on the public roadmap to communicate work in
+  progress.
+- **module design RFC [optional]** - Request for comment documents will be created for large / complex designs that
+  require explanation and robust discussion. RFCs are stored in a separate GitHub repo, and the top-level tracking issue
+  should include link to the RFC.
+- **feature requests** - we have a backlog of feature requests tagged for each module. When we decide to include a
+  customer feature request in the scope of work, links to these issues are added to the module’s tracking issue.
+  Customers can +1 and comment on individual feature requests, just like they can +1 and comment on the top-level
+  tracking issue. _Note_: it is not a requirement to address all feature requests when maturing a construct library.
+- **bugs** - similar to feature requests, we have a backlog of P1 and P2 bugs tagged for each module. When we decide to
+  fix specific, known bugs as part of maturing a construct, links to these issues are added to the module’s tracking
+  issue.
+  > **Note**: it is not a requirement to address all P1 or P2 bugs when maturing a construct library.
+
+![GitHubArtifactsPNG](../images/GitHubArtifacts.png)
+
+# Appendix B: Module complexity T-shirt sizes
+
+One of our tenets is to prioritize work on large/complex modules to ensure we make progress on high-level abstractions
+for services that we cannot complete in a single quarter.Measured in the amount of time required to graduate a module
+from CFN-only to a Developer Preview release candidate, the proposed sizing scale is:
+
+- **Small**: 1 month or less
+- **Medium**: 2-3 months
+- **Large**: 3-5 months
+- **X-large**: 6 months or more
+
+# Appendix C: Action Items
```

Comment:
Take another pass and identify opportunities to automate, particularly when some action is performed or threshold is reached; and we don't realize that it has happened. 
For example, a new module is created as CFN only because of new CloudFormation artifacts. This happens automatically, so we'll need some way of notifying somewhere to add the activities to be performed around it. Similarly, there may be metrics that we don't currently track that may be part of a checklist, etc.

Reply:
OK, I've made a few edits to address this. At this point I think this RFC is serviceable. Let's go through a couple construct module graduation cycles and then refine based on what we learn.